setLocation.go

```diff
@@ -0,0 +1,109 @@
+package main
+
+import (
+	"errors"
+	"encoding/json"
+	"fmt"
+	"github.com/go-telegram-bot-api/telegram-bot-api"
+	"log"
+	"net/http"
+	"net/url"
+	"strconv"
+	"strings"
+)
+
+// Return from www.cepaberto.com
+type cepResponse struct {
+	CEP        string  `json:"cep"`
+	Logradouro string  `json:"logradouro"`
+	Bairro     string  `json:"bairro"`
+	Cidade     string  `json:"cidade"`
+	Estado     string  `json:"estado"`
+	Latitude   string  `json:"latitude"`
+	Longitude  string  `json:"longitude"`
+	Altitude   float64 `json:"altitude"`
+	DDD        int     `json:"ddd"`
+	IBGE       string  `json:"ibge"`
+}
+
+// locationHandler receive postal code from user.
+func locationHandler(update tgbotapi.Update, bot *tgbotapi.BotAPI, config botConfig, msgs botMessages) error {
+
+	tmp := strings.ToLower(strings.Trim(update.Message.CommandArguments(), " "))
+	tmp = strings.Replace(tmp, "-", "", -1)
+	tmp = strings.Replace(tmp, ".", "", -1)
+	tmp = strings.Replace(tmp, "/", "", -1)
+	args := strings.Split(tmp, " ")
+	user := update.Message.From
+	cep := ""
+
+	cmd := strings.ToLower(update.Message.Command())
+
+	if cmd == "setlocation" && len(args) != 2 {
+		return errors.New("/setlocation <pais> <código postal>");
+	}
+
+	if cmd == "cep" && len(args) != 1 {
+		return errors.New("/cep <código postal>");
+	}
+
+	if  (cmd == "setlocation" && args[0] == "br") || cmd == "cep" {
+		if cmd == "setlocation" {
+			cep = args[1]
+		} else {
+			cep = args[0]
+		}
+
+		if err := findCEP(user,cep,config); err != nil {
+			return errors.New("Não foi possível achar a sua localização . CEP "+cep);
+		}
+
+		return errors.New(msgs.LocationSuccess);
+	}
+
+	return errors.New("Não sei como procurar o Código Postal deste país ("+args[0]+")");
```

Comment:
This may be a better as:
fmt.Errorf("não sei como procurar o código postal desse país: %q", args[0])
but see above for the complete change of this block.